twake/backend/core/src/Twake/Discussion/Entity/Message.php

```diff
@@ -478,8 +478,106 @@ public function setIncrementAtTime($increment_at_time)
+        
+        $orignal_files = $messageEntity->getContent()['files'];
+
+        foreach($orignal_files as $file){
+            if($file["type"] == "file") {
+                $new_file_format = Array(
+                    "company_id" => "",
+                    "message_id" => $messageEntity->getId(),
+                    "id" => $file["content"],
+                    "metadata" => Array(
+                        "source" => "drive",
+                        "external_id" => $file["content"]
+                    ),
+                );
+                array_push($files, $new_file_format);
+            }
+        }
+
+        return $files;
+    }
+    
+    /**
+     * Set blocks for new api object
+     */
+    public function setBlocks($messageEntity) {
+        $blocks = Array();
+        $content = $messageEntity->getContent();
+
+        if(!isset($content)) return $blocks;
+
+        $markdown_element = Array(
+            "type" => "mrkdwn",
+            "text" => isset($content['original_str']) ? $content['original_str'] : ""
+        );
+
+        $new_block_format = Array(
+            "type" => "section",
+            "text" => $markdown_element
+        );
+
+        array_push($blocks, $new_block_format);
+
+        return $blocks;
+    }
+
+    /**
+     * Generate new api object
+     */
+    public function generateNewApiObject($messageEntity, $array) {
+    
+        return $array = Array(
+            "id" => $messageEntity->getId(),
+            "channel_id" => $messageEntity->getChannelId(),
+            "thread_id" => $messageEntity->getParentMessageId(),
+            "created_at" => ($messageEntity->getCreationDate() ? $messageEntity->getCreationDate()->getTimestamp() : null),
+            "type" => $messageEntity->getStringMessageType(),
+            "subtype" => null, // TODO
+            "application_id" => $messageEntity->getApplicationId(),
+            "user_id" => ($messageEntity->getSender() ? $messageEntity->getSender()->getId() : null),
+            "edited" => $messageEntity->getEdited(),
+            "text" => isset($messageEntity->getContent()['original_str']) ? $messageEntity->getContent()['original_str'] : "", 
+            "blocks" => $messageEntity->setBlocks($messageEntity),
+            "files" => $messageEntity->setFiles($messageEntity),
+            "context" => $messageEntity->getHiddenData(),
+            "title" => $messageEntity->getHiddenData()['custom_title'] ?: null,
+            "picture" => $messageEntity->getHiddenData()['custom_icon'] ?: null,
+            "stats" => Array(
+                "answers" => $messageEntity->getResponsesCount()
+            ),
+            "pinned_info" => Array(
+                "pinned_by" => ($messageEntity->getSender() ? $messageEntity->getSender()->getId() : null), // TODO
+                "pinned_at" => 0,
+            ),
+            "reactions" => $messageEntity->getReactions(), // TODO Change Reactions format 
```

Comment:
You still need to change the format then ?

Reply:
Reactions are currently formatted in a different way:
![Capture d’écran de 2021-04-26 16-40-22](https://user-images.githubusercontent.com/36481167/116101467-1f9f8380-a6ae-11eb-8473-86db5e05819d.png)

Maybe I could change that if you want ? 😄

Reply:
Yes, it is part of the PR goal to change also this format :)

Reply:
So instead of having a list, just add the name parameter to the list

Reply:
Updated :)